src/main.rs

```diff
@@ -530,36 +529,43 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
         .into());
     }
 
-    let config_directory = solana_root.get_root_path().join("config-k8s");
-    let mut genesis = Genesis::new(config_directory.clone(), genesis_flags);
+    let no_bootstrap = matches.is_present("no_bootstrap");
+    let config_directory = cluster_data_root.get_root_path().join("config-k8s");
+    let retain_previous_genesis = no_bootstrap;
+    let mut genesis = Genesis::new(
+        config_directory.clone(),
+        genesis_flags,
+        retain_previous_genesis,
+    );
 
-    genesis.generate_faucet()?;
-    info!("Generated faucet account");
+    if !no_bootstrap {
+        genesis.generate_faucet()?;
+        info!("Generated faucet account");
 
-    genesis.generate_accounts(ValidatorType::Bootstrap, 1)?;
-    info!("Generated bootstrap account");
+        genesis.generate_accounts(ValidatorType::Bootstrap, 1, None)?;
+        info!("Generated bootstrap account");
 
-    genesis.create_client_accounts(
-        client_config.num_clients,
-        &client_config.bench_tps_args,
-        DEFAULT_CLIENT_LAMPORTS_PER_SIGNATURE,
-        &config_directory,
-        &deploy_method,
-        solana_root.get_root_path(),
-    )?;
-    info!("Client accounts created");
+        genesis.create_client_accounts(
+            client_config.num_clients,
+            &client_config.bench_tps_args,
+            DEFAULT_CLIENT_LAMPORTS_PER_SIGNATURE,
+            &config_directory,
+            cluster_data_root.get_root_path(),
+        )?;
+        info!("Client accounts created");
 
-    // creates genesis and writes to binary file
-    genesis
-        .generate(solana_root.get_root_path(), &build_path)
-        .await?;
-    info!("Genesis created");
+        // creates genesis and writes to binary file
+        genesis
+            .generate(cluster_data_root.get_root_path(), &exec_path)
+            .await?;
+        info!("Genesis created");
+    }
 
     // generate standard validator accounts
-    genesis.generate_accounts(ValidatorType::Standard, num_validators)?;
+    genesis.generate_accounts(ValidatorType::Standard, num_validators, Some(&image_tag))?;
     info!("Generated {num_validators} validator account(s)");
 
-    genesis.generate_accounts(ValidatorType::RPC, num_rpc_nodes)?;
+    genesis.generate_accounts(ValidatorType::RPC, num_rpc_nodes, Some(&image_tag))?;
```

Comment:
It doesn't need to be done now, but it's a bit confusing that this calls `genesis.generate_accounts` *after* genesis has been created.

`generate_accounts` could potentially be refactored somewhere else which just handles keypair files rather than genesis-related things.